prov/shm/src/smr_atomic.c

```diff
@@ -116,53 +116,6 @@ static void smr_format_inject_atomic(struct smr_cmd *cmd,
 	}
 }
 
-static int smr_fetch_result(struct smr_ep *ep, struct smr_region *peer_smr,
-			    struct iovec *iov, size_t iov_count,
-			    const struct fi_rma_ioc *rma_ioc, size_t rma_count,
-			    enum fi_datatype datatype, size_t total_len)
-{
-	int ret, i;
-	struct iovec rma_iov[SMR_IOV_LIMIT];
-
-	for (i = 0; i < rma_count; i++) {
-		rma_iov[i].iov_base = (void *) rma_ioc[i].addr;
-		rma_iov[i].iov_len = rma_ioc[i].count * ofi_datatype_size(datatype);
-	}
-
-	ret = process_vm_readv(peer_smr->pid, iov, iov_count,
-			       rma_iov, rma_count, 0);
-	if (ret != total_len) {
-		if (ret < 0) {
-			FI_WARN(&smr_prov, FI_LOG_EP_CTRL,
-				"CMA write error\n");
-			return -errno;
-		} else { 
-			FI_WARN(&smr_prov, FI_LOG_EP_CTRL,
-				"partial read occurred\n");
-			return -FI_EIO;
-		}
-	}
-
-	return 0; 
-}
-
-static void smr_post_atomic_resp(struct smr_ep *ep, struct smr_cmd *cmd,
-				 const struct iovec *result_iov, size_t count)
-{
-	struct smr_tx_entry *pend;
-	struct smr_resp *resp;
-
-	assert(!ofi_cirque_isfull(smr_resp_queue(ep->region)));
-	resp = ofi_cirque_tail(smr_resp_queue(ep->region));
-
-	cmd->msg.hdr.data = (uintptr_t) ((char **) resp -
-			    (char **) ep->region);
-
-	pend = freestack_pop(ep->pend_fs);
-	smr_format_pend_resp(pend, cmd, NULL, result_iov, count, resp);
-	ofi_cirque_commit(smr_resp_queue(ep->region));
-}
-
```

Comment:
Please update the commit title to indicate that this is fixing a potential data corruption/ordering issue, so that it's clear we may want to backport it as a bug fix.  And on that note, do we?

Reply:
Ah sorry just saw this comment. Probably should back port this fix. I'll open a PR.